packages/compiler/core/types.ts

```diff
@@ -50,7 +50,18 @@ export interface DecoratedType {
  */
 export type TemplatedType = Model | Operation | Interface | Union;
 
+export interface TypeMapper {
+  partial: boolean;
+  getMappedType(type: TemplateParameter): Type;
+  args: readonly Type[];
+  /** @internal */ map: Map<TemplateParameter, Type>;
+}
+
 export interface TemplatedTypeBase {
+  templateMapper?: TypeMapper;
+  /**
+   * @deprecated use templateMapper instead.
+   */
   templateArguments?: Type[];
```

Comment:
Any opinions on what we should do here about that list. Deprecate it or just remove it?

I guess it shouldn't affect emitter and libraries that don't  see templated operation in interface if we keep it deprecated.

Reply:
Looks like it's pretty easy to adapt existing uses of `templateArguments` in linter and decorators which check if a type is templated (check `type.templateMapper?.partial` instead) so I'm OK with deprecating it.

Reply:
Should be using `isTemplateInstance` anyway right? Good time to upgrade.

My feeling though is that we should leave this in but deprecated for a while. I fear our Morgan Stanley friends will be using this. I wonder if we could turn it into a getter that logs a deprecation warning, even?